crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -264,10 +280,18 @@ impl Format<PyFormatContext<'_>> for FormatDanglingOpenParenthesisComments<'_> {
                 "Expected dangling comment to be at the end of the line"
             );
 
+            let slice = comment.slice();
             write!(
                 f,
                 [
-                    line_suffix(&format_args!(space(), space(), format_comment(comment))),
+                    line_suffix(
+                        &format_args!(space(), space(), format_comment(comment)),
+                        // Marking the comment as a line suffix with reserved width is safe since we expect the comment to be end of line.
+                        measure_text(
+                            &f.context().source()[slice.range()],
```

Comment:
Uh, this is a bit more complicated than I thought it is. Computing the width of the source text is correct in most cases, but not when `format_comment` normalizes the comment. We may need to extract the formatting logic of `format_comment` into a `write_comment` function that returns `FormatResult<u32>` where the `u32` is the length of the formatted comment.

Reply:
Hmm. I see. Makes sense.

Just so I understand better, this means there's no way to track what's written to the buffer on each write operation (as the caller)? If not, would it makes sense to have this kind of abstraction more generally in case you ever want to know how much data was written to the buffer on any write operation?

Originally I was wondering if there was a way to write the data to some intermediate buffer, and then push it to the formatter's buffer. I was hoping I could do that so I could determine exactly how much data is written for each formatted element.

Reply:
> We may need to extract the formatting logic of format_comment into a write_comment

Is this possible without a throwaway format context (and a redundant write)? Wouldn't `line_suffix` still need to be provided `FormatComment`? If I'm not missing something, would it be better to have some method on `FormatComment` that gives you the length of the formatted comment? So something like

```rust

let formatted_comment = format_comment(trailing);
let comment_len = formatted_comment.len(); // Determines length with relevant normalizing logic.
write!(
    f,
    [
        line_suffix(
            &format_args![space(), space(), formatted_comment],
            comment_len + 2 // Account for two added spaces
            ),
            expand_parent()
    ]
)?;
```

Reply:
I've added `normalize_comment` without any additional refactor (stepping away; I'll come back). 6726551234e944c44b5951dca26e24d12e5cc662

Reply:
I decided to go with the `write_comment` idea since I was having trouble refactoring the normalization into a state I was happy with. I didn't want to submit something where there were multiple places to maintain normalization logic.

3186ff15c960668e2f25ae15e8027b1596df99c2